### ISSUE TITLE
Too short time to do a manual proof.

### ISSUE THREAD

**Issue by sebastiannielsen** [keybase/keybase-issues#1753]
I feel that the time to do a PGP key proof is too short. I Think the time should be extendable, preferable with a CSRF protected button, to prevent attacks. I understand that there is a need for a timelimit when proofing for the security, but the current timelimit is too short and halfway through the process I get that the time expired.

**Comment by thecoshman** [comment]
How long is this time limit currently? How long do you propose it should be?

**Comment by sebastiannielsen** [comment]
As I wrote, I do not propose that the time limit should be extended, instead I propose that there should be a means to reset the counter/time while doing the manual proof. Eg some button that tells the system "Im still here". Preferable with a on-screen timer that tells how much time its left.

I don't know exactly how long the time limit is, but it seems to be 5 minutes.

The time limit is there to prevent certain CSRF, XSS or session attacks. By allowing the end user to renew the session with a button that is sufficently protected against attacks, the user can take ample time to do the proof without reducing security.

**Comment by thecoshman** [comment]
We are talking about doing tasks like generating a special tweet and posting it, right? Because I think five minutes is more than enough time by far, even considering those who might not able to use a computer that fast for various reasons.

**Comment by sebastiannielsen** [comment]
No, im talking about submitting the signed entity to keybase. There is 3 options: In the browser, Command line with keybase, or manual signing with OpenPGP.

When you select the manual signing, you are asked to run a command on the key hosting computer. The time given to run this command is too short, since I had to convert the data to a java object, to then import the entity-to-be-signed into a client application that uses java, and then get the signed entity back. Then I had to run the other half on the command, on the now signed entity.

I tried then to do it directly over a VNC remote solution with the server in a terminal window but the VNC remote solution does not support copy+paste on the server side, so typing the whole command takes too long time too.

Thats why there needs to be some button so you can regarly "refresh" the session with the command to type to sign & upload the object to keybase. Such refresh button should not induce any security related difficulties, since the button can be sufficently be protected with a CSRF token.

The only solution left was to export the private key, import this into keybase, then signing the object in browser, and then deleting the key from keybase.

**Comment by jrootabega** [comment]
I second the need for some improvement here. I am trying add a strictly offline key to my account, and the process to get it transferred to my airgapped signing system is cumbersome enough that it times out before I can reasonably complete it. The process of copying the signature request data, getting it over to the signing system, running the signing command, getting the signature back to the online system, and then plugging the signature back into the curl command can be a pain. I haven't made it yet even if rushing. I'm fully able-bodied and able-minded, and lucky enough to have a laptop on the other table I can walk right over to. Not everyone will be so lucky.
